code/game/objects/items/devices/flashlight.dm

```diff
@@ -238,52 +238,41 @@
 	name = "flare"
 	desc = "A red USCM issued flare. There are instructions on the side, it reads 'pull cord, make light'."
 	w_class = SIZE_SMALL
-	brightness_on = 5 //As bright as a flashlight, but more disposable. Doesn't burn forever though
+	brightness_on = 4
 	icon_state = "flare"
 	item_state = "flare"
 	actions = list()	//just pull it manually, neckbeard.
 	raillight_compatible = 0
-	var/fuel = 0
-	var/fuel_rate = AMOUNT_PER_TIME(1 SECONDS, 1 SECONDS)
+	var/flare_time = 1 MINUTES
+	var/flare_variance = 15 SECONDS // 15 seconds one way, 15 seconds the other way
+	var/burned_out = FALSE
+	var/burnout_timer_id
 	var/on_damage = 7
 	var/ammo_datum = /datum/ammo/flare
 
-/obj/item/device/flashlight/flare/Initialize()
-	. = ..()
-	fuel = rand(1600 SECONDS, 2000 SECONDS)
-
 /obj/item/device/flashlight/flare/dropped(mob/user)
 	. = ..()
 	if(iscarbon(user) && on)
 		var/mob/living/carbon/flare_user = user
 		flare_user.toggle_throw_mode(THROW_MODE_OFF)
 
-/obj/item/device/flashlight/flare/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	return ..()
-
-/obj/item/device/flashlight/flare/process(delta_time)
-	fuel -= fuel_rate * delta_time
-	if(fuel <= 0 || !on)
-		burn_out()
-
 /obj/item/device/flashlight/flare/proc/burn_out()
 	turn_off()
-	fuel = 0
+	burned_out = TRUE
 	icon_state = "[initial(icon_state)]-empty"
 	add_to_garbage(src)
-	STOP_PROCESSING(SSobj, src)
 
 /obj/item/device/flashlight/flare/proc/turn_on()
 	on = TRUE
 	heat_source = 1500
 	update_brightness()
 	force = on_damage
 	damtype = "fire"
-	START_PROCESSING(SSobj, src)
+	var/burn_time = flare_time + rand(flare_variance * -1, flare_variance)
+	burnout_timer_id = addtimer(CALLBACK(src, .proc/burn_out), burn_time, TIMER_UNIQUE|TIMER_STOPPABLE)
```

Comment:
deltimer() in Destroy() too, refs can be kept for a while if flare is destroyed (explosions etc)